### PR TITLE
Add configurable list ordering (created_at | updated_at)

### PR DESCRIPTION
**Pull request by arthur-debert**
## Summary

- Adds a new `ordering` config key with values `created_at` (default, unchanged) and `updated_at`. Under `updated_at`, listings show the most-recently-modified pad first.
- When a nested pad is edited, `updated_at` bubbles up the parent chain (new `propagate_modification` helper, mirrored on `propagate_status_change`), so an ancestor surfaces to the top under `updated_at` ordering instead of leaving it buried.
- Plumbed via a thread-local set by `initialize()` after config load, so every call path through `indexed_pads` picks up the configured key without threading the parameter through ~20 call sites. `index_pads` still accepts `OrderingKey` explicitly for unit tests.

## Changes

- `config.rs`: `OrderingKey` enum (`CreatedAt` | `UpdatedAt`) + `ordering` field with confique default. Tests for (de)serialization, default, and display.
- `index.rs`: thread-local `CURRENT_ORDERING`, `set_ordering_key` / `current_ordering_key`, sort dispatch in `index_level`. Unit tests for both keys, including recursive children.
- `todos.rs`: new `propagate_modification` helper (stamps `Utc::now()` on each ancestor, stops at missing parent). Unit tests including the missing-parent edge case.
- `commands/update.rs`: calls `propagate_modification` after saving the edited pad.
- `init.rs`: publishes `config.ordering` to the thread-local after load.
- `tests/ordering_e2e.rs`: four end-to-end cases exercising the full binary via `cargo_bin`:
  1. Default `created_at` ordering.
  2. `updated_at` reorders after an edit.
  3. Toggling back to `created_at` restores the creation sequence.
  4. Editing a nested child bubbles the parent to the top under `updated_at`.

## Design notes

- Display Indexes are not comparable across ordering switches (pad "3" under `created_at` is a different pad than "3" under `updated_at`). This is expected for a shell tool where indexes reflect the current view.
- Thread-local chosen over explicit parameter threading to keep the blast radius small; tests that want non-default ordering call `set_ordering_key` or pass `OrderingKey` directly to `index_pads`.

## Test plan

- [x] `cargo test` — all existing tests + new unit + e2e tests pass (600+ tests green).
- [x] `cargo clippy --all-targets -- -D warnings` — clean.
- [x] `cargo fmt --check` — clean.
- [x] Manual: `padz config set ordering updated_at`, edit a nested child, confirm ancestor surfaces in `padz list`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/padzapp/src/todos.rs [hunk 1]
```diff
@@ -125,6 +125,36 @@ pub fn propagate_status_change<S: DataStore>(
...
+        let mut parent_pad = match store.get_pad(&parent_id, scope, Bucket::Active) {
+            Ok(p) => p,
+            Err(_) => break,
+        };
+
+        parent_pad.metadata.updated_at = now;
+        store.save_pad(&parent_pad, scope, Bucket::Active)?;
+        current_parent_id = parent_pad.metadata.parent_id;
+    }
+
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/padzapp/src/todos.rs]: Good catch — this is the root issue, and addressed in 8fc66c8. The implementation pivots to compute effective `updated_at` at index time via the new `effective_updated_at` helper in `index.rs`. `metadata.updated_at` is now reserved as the file-mtime proxy as you described; `propagate_modification` and its call from `update::run` are removed. The nested-edit-surfaces-parent behavior is preserved (see `test_ordering_updated_at_surfaces_parent_via_descendant_edit` and the e2e test) without ever mutating ancestor state.

File: crates/padzapp/src/todos.rs [hunk 2]
```diff
@@ -125,6 +125,36 @@ pub fn propagate_status_change<S: DataStore>(
...
+    scope: Scope,
+    child_parent_id: Option<Uuid>,
+) -> Result<()> {
+    let mut current_parent_id = child_parent_id;
+    let now = chrono::Utc::now();
+
+    while let Some(parent_id) = current_parent_id {
+        let mut parent_pad = match store.get_pad(&parent_id, scope, Bucket::Active) {
+            Ok(p) => p,
+            Err(_) => break,
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/padzapp/src/todos.rs]: Moot in 8fc66c8 — `propagate_modification` is removed entirely (effective `updated_at` is computed at index time, see reply on #3140481144). Worth noting that `propagate_status_change` above has the same `Err(_) => break` pattern; I left that alone since it predates this PR and the same fix would apply equally there.

File: crates/padzapp/src/index.rs [hunk 3]
```diff
@@ -202,9 +238,13 @@ pub fn index_pads(active: Vec<Pad>, archived: Vec<Pad>, deleted: Vec<Pad>) -> Ve
...
     mut pads: Vec<TaggedPad>,
     parent_map: &HashMap<Option<Uuid>, Vec<TaggedPad>>,
+    ordering: OrderingKey,
 ) -> Vec<DisplayPad> {
-    // Sort by created_at descending (newest first) within this level
-    pads.sort_by(|a, b| b.pad.metadata.created_at.cmp(&a.pad.metadata.created_at));
+    // Sort descending (newest first) within this level, keyed per config.
+    pads.sort_by(|a, b| match ordering {
+        OrderingKey::CreatedAt => b.pad.metadata.created_at.cmp(&a.pad.metadata.created_at),
+        OrderingKey::UpdatedAt => b.pad.metadata.updated_at.cmp(&a.pad.metadata.updated_at),
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/padzapp/src/index.rs]: Addressed in 8fc66c8. The sort comparator now applies the suggested tie-breaker chain (timestamp → other timestamp → UUID) for both ordering keys. Locked in by `test_ordering_tie_break_is_deterministic`, which sorts the same two pads with identical timestamps in both input orders and asserts identical output.

File: crates/padzapp/src/index.rs [hunk 4]
```diff
@@ -153,7 +182,14 @@ struct TaggedPad {
...
 /// Each entry's `children` vector follows the same ordering recursively.
-pub fn index_pads(active: Vec<Pad>, archived: Vec<Pad>, deleted: Vec<Pad>) -> Vec<DisplayPad> {
+///
+/// `ordering` controls the sort key applied within each bucket and at each nesting level.
+pub fn index_pads(
+    active: Vec<Pad>,
+    archived: Vec<Pad>,
+    deleted: Vec<Pad>,
+    ordering: OrderingKey,
+) -> Vec<DisplayPad> {
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/padzapp/src/index.rs:192]: Acknowledged but deliberately not taken — the repo's CLAUDE.md is explicit: "NO BACKWARDS COMPATIBILITY NOR ADAPTERS NOR DEPRECATED ANYTHING NOT FALLBACKS unless explicitly requested". `padzapp` is a workspace internal crate (the only public consumer is the `padz` CLI in the same workspace), so the breaking signature change has no external blast radius and matches the repo's preference for explicit, churn-on-change APIs over compat shims.
